crates/diesel_models/src/query/capture.rs

```diff
@@ -0,0 +1,116 @@
+use diesel::{associations::HasTable, BoolExpressionMethods, ExpressionMethods, Table};
+use router_env::{instrument, tracing};
+
+use super::generics;
+use crate::{
+    capture::{Capture, CaptureNew, CaptureUpdate, CaptureUpdateInternal},
+    enums::CaptureStatus,
+    errors,
+    schema::captures::dsl,
+    PgPooledConn, StorageResult,
+};
+
+impl CaptureNew {
+    #[instrument(skip(conn))]
+    pub async fn insert(self, conn: &PgPooledConn) -> StorageResult<Capture> {
+        generics::generic_insert(conn, self).await
+    }
+}
+
+impl Capture {
+    #[instrument(skip(conn))]
+    pub async fn find_optional_by_payment_id_merchant_id(
+        conn: &PgPooledConn,
+        payment_id: &str,
+        merchant_id: &str,
+    ) -> StorageResult<Option<Self>> {
+        generics::generic_find_one_optional::<<Self as HasTable>::Table, _, _>(
+            conn,
+            dsl::merchant_id
+                .eq(merchant_id.to_owned())
+                .and(dsl::payment_id.eq(payment_id.to_owned())),
+        )
+        .await
+    }
+    #[instrument(skip(conn))]
+    pub async fn find_by_payment_id_merchant_id(
```

Comment:
capture is associated with payment attempt, so we need payment_attempt_id and capture_id to do a find one, in which case we are using this query? find_optional_by_payment_id_merchant_id also falls under the same condition.

Reply:
what will be the use case to find a capture by `capture_id`, and how do we get `capture_id` to query the child capture? as I see, it is queried by `autorized_attempt_id` and always returns a list of all the captures